server.js

```diff
@@ -51,37 +61,72 @@ let messageList = [];
 const addToList = message => messageList.push(message);
 subscribers.push(addToList);
 
-// https://expressjs.com/en/starter/basic-routing.html
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/views/index.html");
-});
+const routeGetRootWithUserId = config => (request, response) => {
+  let template = PageWrapper({
+    title: config.pageTitle,
+    children: ChatRoom({
+      userName: request.params.id ? request.params.id : config.defaultUserName
+    })
+  });
+  response.send(template);
+}
 
-app.get("/messages", (request, response) => {
-  response.type("text/plain");
+const routeGetRoot = config => (request, response) => {
+  let template = PageWrapper({
+    title: config.pageTitle,
+    children: SignIn()
+  });
+  response.send(template);
+}
+
+const routePostRoot = config => (request, response) => {
+  response.redirect(`/${request.body.user}`);
+}
+
+const routeGetMessages = config => (request, response) => {
+  response.type('text/html');
   const write = sentence => {
-    const line = `${sentence}\n`;
+    const line = `${sentence}`;
     response.write(line);
   };
-
-  subscribers.push(write);
   const cutoff = Math.max(messageList.length - 10, 0);
-
   const latest = messageList.slice(cutoff);
   if (cutoff !== 0) {
     messageList = latest;
   }
-
+  subscribers.push(write);
   messageList.forEach(write);
-});
+};
 
-app.post("/messages", (request, response) => {
+const routePostMessages = config => (request, response) => {
```

Comment:
we curry all the routes to accept a config object. The function will return (curry) another function with the prepared configuration.